src/lib/artefact.ml

```diff
@@ -1,90 +1,89 @@
+  let ranking = Ranking
+  let yans criterion = Yans { criterion; }
+
+end
+
+(* Artefact values *)
+(* ************************************************************************* *)
+
+type bonus = int
+(* Bonus value *)
+
+type rank = int
+(* ranks, from 1 to <n> (for some <n>) *)
+
+type yan =
+  | Yes
+  | Alt
+  | No (**)
+(* Yes/Alt/No *)
+
 type t =
-  | Rank of int
-  | Note of {
-      technique : int;
-      musicality : int;
-      teamwork : int;
-    }
-  | Single_note of int
-
-let to_string = function
-  | Rank i -> string_of_int i
-  | Single_note i -> Format.asprintf "%d-" i
-  | Note { technique; musicality; teamwork; } ->
-    Format.asprintf "%d-%d-%d" technique musicality teamwork
-
-let of_string s =
-  match int_of_string s with
-  | res -> Rank res
-  | exception Failure _ ->
-    begin match String.split_on_char '-' s with
-      | [note; _] ->
-        Single_note (int_of_string note)
-      | [technique; musicality; teamwork] ->
-        Note {
-          technique = int_of_string technique;
-          musicality = int_of_string musicality;
-          teamwork = int_of_string teamwork;
-        }
-      | _ -> failwith "bad encoded artefact"
-    end
-
-let () =
-  State.add_init (fun st ->
-      Sqlite3_utils.exec0_exn st {|
-        CREATE TABLE IF NOT EXISTS judgements (
-          phase INTEGER,
-          judge INTEGER,
-          target TEXT,
-          artefact TEXT,
-        CONSTRAINT unicity
-          UNIQUE (phase, judge, target)
-          ON CONFLICT REPLACE
-        )
-      |})
-
-let conv =
-  Conv.mk Sqlite3_utils.Ty.[text] of_string
-
-let get st ~phase ~judge ~target : t option =
-  let open Sqlite3_utils.Ty in
-  try
-    Some (State.query_one_where ~st ~conv ~p:[int; int; text]
-            {|SELECT artefact FROM judgements WHERE phase = ? AND judge = ? AND target = ?|}
-            phase judge (Target.to_string target))
-  with Sqlite3_utils.RcError Sqlite3_utils.Rc.NOTFOUND -> None
-
-let set st ~phase ~judge ~target = function
-  | None ->
-    let open Sqlite3_utils.Ty in
-    State.insert ~st ~ty:[int; int; text]
-      {|DELETE FROM judgements WHERE phase = ? AND judge = ? AND target = ?|}
-      phase judge (Target.to_string target)
-  | Some artefact ->
-    let open Sqlite3_utils.Ty in
-    State.insert ~st ~ty:[int; int; text; text]
-      {|INSERT INTO judgements(phase,judge,target,artefact) VALUES (?,?,?,?)|}
-      phase judge (Target.to_string target) (to_string artefact)
-
-let list st ~phase ~judge =
-  let open Sqlite3_utils.Ty in
-  let conv =
-    Conv.mk [text; text]
-      (fun target artefact -> (Target.of_string target, of_string artefact))
+  | Bonus of bonus
+  | Rank of Rank.t
+  | Yans of yan list
+
+
+(* DB interaction *)
+(* ************************************************************************* *)
+
+(* Int encoding schema:
+
+   An integer encoding an artefact can only be decoded if the kind of the
+   artefact is provided (i.e. it is a tagless encoding). *)
```

Comment:
Add comment to why use this int encoding schema
```
We use this encoding schema to store an artefact into a unique row in the database.
It avoids the need for several rows in case of an arbitrary number of criterion for the artefact. This case is expected when `kind` matches `Yes_Alt_No List`.
The maximum number of criterion with this schema is * (4?) because an int use * (1?) bytes.
```